sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/WriteToDataSourceV2Exec.scala

```diff
@@ -347,10 +347,26 @@ trait BatchWriteHelper {
   }
 }
 
+/**
+ * A base trait for executing a write eagerly and caching the result, similar to
+ * `ExecutedCommandExec`, so that a write is executed only once.
+ */
+trait V2TableWriteExecBase extends SparkPlan {
```

Comment:
can we reuse `V2CommandExec`? The major difference is the return type of `run`, but I don't think there is a command that must return an RDD.